casper/src/main/scala/coop/rchain/casper/BlockValidationLogic.scala

```diff
@@ -113,6 +113,28 @@ object BlockValidationLogic {
       .getOrElse(BlockStatus.valid.asRight[InvalidBlock])
   }
 
+  def transactionExpiration(
+      b: BlockMessage,
+      expirationThreshold: Int,
+      expiredDeployF: String => Unit
+  ): ValidBlockProcessing = {
+    val earliestAcceptableValidAfterBlockNumber = b.blockNumber - expirationThreshold
+    val deploys                                 = b.state.deploys.map(_.deploy)
+    val maybeExpiredDeploy =
+      deploys.find(_.data.validAfterBlockNumber <= earliestAcceptableValidAfterBlockNumber)
+    maybeExpiredDeploy
+      .map { expiredDeploy =>
+        expiredDeployF(
+          ignore(
+            b,
+            s"block contains an expired deploy with valid after block number of ${expiredDeploy.data.validAfterBlockNumber}: ${expiredDeploy.data.term}"
+          )
+        )
```

Comment:
This is very confusing to me. Would be nice to get rid of this `expiredDeployF: String => Unit`. Its not clear what is the purpose of it.